cpp/nacjd.cc

```diff
@@ -0,0 +1,371 @@
+/** \file    nacjd.cc
+    \brief   Identifies URL's that we can use for further processing..
+    \author  andreas-ub
+
+    \copyright 2021 Universitätsbibliothek Tübingen
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <iostream>
+#include <unordered_set>
+#include <utility>
+#include <cerrno>
+#include <cstring>
+#include <getopt.h>
+#include <sstream>
+#include <iterator>
+#include <algorithm>
+#include <fstream>
+#include "util.h"
+#include "FileUtil.h"
+#include "HttpHeader.h"
+#include "UrlUtil.h"
+#include "JSON.h"
+#include "MARC.h"
+#include "UBTools.h"
+#include "Downloader.h"
+
+
+namespace {
+
+
+[[noreturn]] void Usage() {
+    ::Usage("marc_title_in_file new_marc_title_out_file\n"
+            "where marc_title_in_file contains also icpsr records (001 or 035a)\n"
+            "these records are not processed any more\n"
+            "new_marc_title_out_file contains all icpsr records not contained in input file.");
+}
+
+int state = 0;
+std::string currentId_website;
+std::vector<std::string> ids_website;
+const unsigned int timeout = 15;
+const std::string nacjd_titles = UBTools::GetTuelibPath() + "nacjd_titles.html";
+const std::string nacjd_new_titles_json = UBTools::GetTuelibPath() + "nacjd_new_titles.json";
+
+
+bool containsValue(std::map<std::string,std::string> &map, std::string search_value) {
+    for (auto const& [key, val] : map) {
+        if (val == search_value)
+            return true;
+    }
+    return false;
+}
+
+
+void handleChar(const char& c)
```

Comment:
Ampersand on wrong side of space and the opening bracket should be on the same line as the function signature unless the line would be too long.  And, no camel case for function names, please.